### PR TITLE
Java.Util.PriorityQueue

### PR DESCRIPTION
**Pull request by truongchauhien**

### CODE REVIEW

File: java/lang/ClassCastException/ClassCastExceptionTest.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// Constructs a new ClassCastException with the specified detail message.
+    ClassCastException classCastExceptionWithMessage = ClassCastException("ClassCastException with the specified message");
+    ASSERT_STR("ClassCastException with the specified message", classCastExceptionWithMessage.getMessage().toString());
+}
+
+TEST(JavaLang, ClassCastExceptionTryCatch) {
+    try {
+        throw ClassCastException("Throw ClassCastException");
+    }
+    catch (Exception ex) {
```
Comment (dthongvl) [inline java/lang/ClassCastException/ClassCastExceptionTest.cpp]: Please catch ClassCastException

File: java/util/PriorityQueue/PriorityQueueTest.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    PriorityQueue<int> initializerListConstructor {1, 2, 3, 4, 5};
+    ASSERT_EQUAL(5, initializerListConstructor.size());
+    ASSERT_EQUAL(5, initializerListConstructor.peek());
+
+    PriorityQueue<int> copyConstructor(initializerListConstructor);
+    ASSERT_EQUAL(5, copyConstructor.size());
+    ASSERT_EQUAL(5, copyConstructor.peek());
+}
+
+TEST(JavaUtil, PriorityQueueDestructor) {
```
Comment (dthongvl) [inline java/util/PriorityQueue/PriorityQueueTest.cpp]: Please remove test if it doesnt have ASSERT statement.

File: java/util/PriorityQueue/PriorityQueue.hpp [hunk 3]
```diff
@@ -22,4 +22,274 @@
...
+             */
+//          PriorityQueue(const Comparator<E> &target);
+
+            /**
+             * Constructor
+             * Creates a PriorityQueue with the specified initial capacity.
+             *
+             * @param initialCapacity
+             */
+//          PriorityQueue(int initialCapacity);
```
Comment (dthongvl) [inline java/util/PriorityQueue/PriorityQueue.hpp]: Why this constructor is disabled?

**Review by dthongvl — CHANGES_REQUESTED** [review]

File: java/util/PriorityQueue/PriorityQueueTest.cpp (hunk 2 above)
Comment (truongchauhien) [reply java/util/PriorityQueue/PriorityQueueTest.cpp]: Done.

File: java/lang/ClassCastException/ClassCastExceptionTest.cpp (hunk 1 above)
Comment (truongchauhien) [reply java/lang/ClassCastException/ClassCastExceptionTest.cpp]: Done.

File: java/util/PriorityQueue/PriorityQueue.hpp (hunk 3 above)
Comment (truongchauhien) [reply java/util/PriorityQueue/PriorityQueue.hpp]: I have enabled it, but I can only test this constructor with `initialCapacity < 1` to throw exception. PriorityQueue class doesn't have "getCapacity" (or similar) method and std::vector<T> doesn't have any method to control exactly initialCapacity.
We use this constructor to ensure that PriorityQueue instance has "enough capacity" (>= initialCapacity argument) as we want.

**Review by dthongvl — APPROVED** [review]

**Pull request merged**